### ISSUE TITLE
erro Unresolved reference: chip

### ISSUE THREAD

**Issue by hainahouwei** [project-chip/connectedhomeip#6094]
#### Problem
<what's wrong or missing, please include any applicable:
 * Click Make Project in Android Studio to build Android CHIPTool
 * Unresolved reference: chip
 >

 #### Proposed Solution
<suggested fix, suggested enhancement>

**Comment by Damian-Nordic** [comment]
@hainahouwei Did you follow the steps described in https://github.com/project-chip/connectedhomeip/tree/master/src/android/CHIPTool? It seems you haven't built the CHIP controller JAR before builting the CHIPTool project.

**Comment by AdamZhang0124** [comment]
Hi @Damian-Nordic @hainahouwei ,
I have encountered the same problem, therefore I would like to check if this issue has any progress?
I do have followed the the steps described in https://github.com/project-chip/connectedhomeip/tree/master/src/android/CHIPTool, but still, the build is failed.

Attach the screenshot fyi:
![image](https://user-images.githubusercontent.com/41885176/124247362-64371780-db54-11eb-9081-96efb5498a53.png)

I noticed that you mentioned **the CHIP controller JAR** , may I know which file/command you were referring to, I did not see it on the README.md?
Many thanks!

**Comment by austinh0** [comment]
`scripts/examples/android_app.sh` should generate `CHIPController.jar` and `SetupPayloadParser.jar`, and copy them to `src/android/CHIPTool/app/libs`.

It looks like you're missing these jars, resulting in the unresolved reference errors. Are you running `scripts/examples/android_app.sh` before you sync/make? Can you check if those jars are there?

**Comment by jonsmirl** [comment]
I just ran into this also. The problem is in the Android build instructions. If the environment is not bootstrapped the android_app.sh script fails.

1) Check out the CHIP repository.

>>>> 1a) Bootstrap it.  source scripts/bootstrap.sh

2) In the command line, run the following command from the top CHIP directory:

TARGET_CPU=arm64 ./scripts/examples/android_app.sh
See the table above for other values of TARGET_CPU.

3) Open the project in Android Studio and run Sync Project with Gradle Files.

Use one of the following options to build an Android package:

4)  Click Make Project in Android Studio.

Run the following command in the command line:

cd src/android/CHIPTool
./gradlew build

**Comment by austinh0** [comment]
Makes sense, bootstrap is required before we build anything in CHIP. I will update the docs.

**Comment by asokfair** [comment]
-->>>> 1a) Bootstrap it. source scripts/bootstrap.sh gives us error, 
![image](https://user-images.githubusercontent.com/6814876/193397341-696866c2-735a-4589-9215-cdf48f6c369b.png)

**Comment by VinhBH1** [comment]
Hi @austinh0 
where is the location of CHIPController.jar and SetupPayloadParser.jar

 i have follow in https://github.com/project-chip/connectedhomeip/blob/master/docs/guides/android_building.md
but still not see  CHIPController.jar and SetupPayloadParser.jar

**Comment by BlackLordj** [comment]
> CHÀO@austinh0 vị trí của CHIPController.jar và SetupPayloadParser.jar ở đâu
> 
> tôi đã theo dõi trong https://github.com/project-chip/connectedhomeip/blob/master/docs/guides/android_building.md nhưng vẫn không thấy CHIPController.jar và SetupPayloadParser.jar
I'm also having the same error as you, I've run the command TARGET_CPU=x64 ./scripts/examples/android_app_ide.sh many times but it still doesn't have .jar files in the android/CHIPtool/app/lib/ folder. Do you have a solution yet?

**Comment by BlackLordj** [comment]
> ```
> jonsmirl@ares:~/aosp/esp-matter/connectedhomeip/connectedhomeip$ ls ./examples/android/CHIPTool/app/libs
> AndroidPlatform.jar  CHIPClusters.jar    jniLibs            libMatterTlv.jar       README.md
> CHIPClusterID.jar    CHIPController.jar  libMatterJson.jar  OnboardingPayload.jar
> ```
I still have the problem that it doesn't create .jar files after running the set_target command
